python/libraries/request.py

```diff
@@ -40,4 +85,41 @@ def _values():
                 [{"code": "int", "response": "str", "return": json_value_type}]
             ),
         ),
+        "delete": (
+            "str",
+            n_map_type.with_typevars(["str", "str"]),
+            n_cmd_type.with_typevars(
+                [{"code": "int", "response": "str", "return": json_value_type}]
+            ),
+        ),
+        "head": (
+            "str",
+            n_map_type.with_typevars(["str", "str"]),
+            n_cmd_type.with_typevars(
+                [{"code": "int", "response": "str", "return": json_value_type}]
+            ),
+        ),
+        "options": (
+            "str",
+            n_map_type.with_typevars(["str", "str"]),
+            n_cmd_type.with_typevars(
+                [{"code": "int", "response": "str", "return": json_value_type}]
+            ),
+        ),
+        "patch": (
+            "str",
+            n_map_type.with_typevars(["str", "str"]),
+            n_map_type.with_typevars(["str", "str"]),
+            n_cmd_type.with_typevars(
+                [{"code": "int", "response": "str", "text": "str"}]
+            ),
+        ),
+        "put": (
+            "str",
+            n_map_type.with_typevars(["str", "str"]),
+            n_map_type.with_typevars(["str", "str"]),
+            n_cmd_type.with_typevars(
+                [{"code": "int", "response": "str", "text": "str"}]
+            ),
+        ),
```

Comment:
Hmm, why not a general request function?

Reply:
Backward compatibility for `request.get` and `request.post`, and also some types need data to send, others don't.

Reply:
Well it can return an error if the request type doesn't allow a body. Just because request.get and request.post exist doesn't mean we need to implement the others for backwards compatibility; we aren't breaking anything by not implementing `request.put`